packages/drivers/routerlicious-driver/src/errorUtils.ts

```diff
@@ -124,8 +125,43 @@ export function throwR11sNetworkError(
 export function errorObjectFromSocketError(
 	socketError: IR11sSocketError,
 	handler: string,
+	additionalProps?: DriverErrorTelemetryProps,
 ): R11sError {
 	// pre-0.58 error message prefix: R11sSocketError
 	const message = `R11s socket error (${handler}): ${socketError.message}`;
-	return createR11sNetworkError(message, socketError.code, socketError.retryAfterMs);
+	return createR11sNetworkError(
+		message,
+		socketError.code,
+		socketError.retryAfterMs,
+		additionalProps,
+	);
+}
+
+/** Simulate the pathname for socket connection */
+export const socketIoPath = "socket.io";
+
+/**
+ * Get a stripped version of a URL safe for r11s telemetry
+ * @returns undefined if no appropriate hostName is provided
+ */
+export function getUrlForTelemetry(hostName: string, path?: string): string | undefined {
+	// Strip off "http://" or "https://"
+	const hostNameMatch = hostName.match(/^(?:https?:\/\/)?([^/]+)/);
+	if (!hostNameMatch) {
+		return undefined;
+	}
+	const strippedHostName = hostNameMatch[1];
+
+	let extractedPath: string | undefined;
+	if (path !== undefined) {
+		// Extract the first portion of the path and explicitly match it to known path names
+		const pathMatch = path.match(/^\/?([^/]+)/);
+		if (pathMatch && [socketIoPath, "repos", "deltas", "documents"].includes(pathMatch[1])) {
```

Comment:
because there are multiple urls that start with the same path segment (i.e. "documents", "repos") and have the same HTTP method we might need some additional information logged to be able to differentiate between those (i.e. total segment length, etc.). @manishgargd to comment if that is really necessary for this task.